Mathlib/Data/Finset/Basic.lean

```diff
@@ -3481,6 +3481,39 @@ theorem toFinset_nonempty_iff (l : List α) : l.toFinset.Nonempty ↔ l ≠ [] :
   simp [Finset.nonempty_iff_ne_empty]
 #align list.to_finset_nonempty_iff List.toFinset_nonempty_iff
 
+theorem toFinset_filter {l : List α} (f : α → Bool):
+    (List.filter f l).toFinset = Finset.filter (f .) (l.toFinset) := by
+  match l with
+  | [] => simp
+  | x :: xs =>
+    rw [List.toFinset_cons]
+    unfold List.filter
+    rw [Finset.filter_insert]
+    cases (f x)
+    simp [toFinset_filter f]
+    simp
+    rw [toFinset_filter f]
+
+theorem toFinset_is_singleton_implies_replicate {l : List α} {a : α}
+    (h : l.toFinset ⊆ {a}) : l = List.replicate l.length a := by
```

Comment:
This lemma follows immediately from [List.eq_replicate_of_mem](https://leanprover-community.github.io/mathlib4_docs/Std/Data/List/Lemmas.html#List.eq_replicate_of_mem).

Reply:
(and therefore should be removed?)